src/ring_theory/graded_algebra/basic.lean

```diff
@@ -77,4 +77,124 @@ def graded_algebra.decompose : A ≃ₐ[R] ⨁ i, 𝒜 i := alg_equiv.symm
   (graded_algebra.decompose 𝒜).symm (direct_sum.of _ i x) = x :=
 direct_sum.submodule_coe_alg_hom_of 𝒜 _ _
 
+/-- The projection maps of graded algebra-/
+def graded_algebra.proj (𝒜 : ι → submodule R A) [graded_algebra 𝒜] (i : ι) : A →ₗ[R] A :=
+(𝒜 i).subtype.comp $
+  (dfinsupp.lapply i).comp $
+  (graded_algebra.decompose 𝒜).to_alg_hom.to_linear_map
+
+@[simp] lemma graded_algebra.proj_apply (i : ι) (r : A) :
+  graded_algebra.proj 𝒜 i r = (graded_algebra.decompose 𝒜 r : ⨁ i, 𝒜 i) i := rfl
+
+lemma graded_algebra.proj_mem (i : ι) (r : A) :
+  graded_algebra.proj 𝒜 i r ∈ 𝒜 i := (graded_algebra.decompose 𝒜 r i).2
+
+/-- The support of `r` is the `finset` where `proj R A i r ≠ 0 ↔ i ∈ r.support`-/
+def graded_algebra.support [Π (i : ι) (x : 𝒜 i), decidable (x ≠ 0)]
+  (r : A) : finset ι :=
+(graded_algebra.decompose 𝒜 r).support
+
+lemma graded_algebra.proj_recompose (a : ⨁ i, 𝒜 i) (i : ι) :
+  graded_algebra.proj 𝒜 i ((graded_algebra.decompose 𝒜).symm a) =
+  (graded_algebra.decompose 𝒜).symm (direct_sum.of _ i (a i)) :=
+by rw [graded_algebra.proj_apply, graded_algebra.decompose_symm_of, alg_equiv.apply_symm_apply]
+
+variable [Π (i : ι) (x : 𝒜 i), decidable (x ≠ 0)]
+
+lemma graded_ring.mem_support_iff
+  (r : A) (i : ι) :
+i ∈ graded_algebra.support 𝒜 r ↔ (graded_algebra.proj 𝒜 i r ≠ 0) :=
+⟨λ hi, begin
+  contrapose! hi,
+  unfold graded_algebra.support,
+  unfold graded_algebra.proj at hi,
+  rw [dfinsupp.mem_support_iff, not_not],
+  simp only [submodule.subtype_apply, dfinsupp.lapply_apply, alg_equiv.to_alg_hom_eq_coe,
+    function.comp_app, linear_map.coe_comp, alg_equiv.to_alg_hom_to_linear_map,
+    alg_equiv.to_linear_map_apply, submodule.coe_eq_zero] at hi,
+  exact hi,
+end, λ hi, begin
+  unfold graded_algebra.proj at hi,
+  unfold graded_algebra.support,
+  simp only [ne.def, dfinsupp.mem_support_to_fun],
+  intro rid,
+  simp only [submodule.subtype_apply, dfinsupp.lapply_apply, alg_equiv.to_alg_hom_eq_coe,
+    function.comp_app, linear_map.coe_comp, alg_equiv.to_alg_hom_to_linear_map,
+    alg_equiv.to_linear_map_apply, submodule.coe_eq_zero] at hi,
+  rw rid at hi,
+  simp only [eq_self_iff_true, not_true, ne.def, submodule.coe_zero] at hi,
+  exact hi,
+end⟩
```

Comment:
A much shorter proof:
```suggestion
@[simp] lemma graded_algebra.mem_support_iff {r : A} {i : ι} :
  i ∈ graded_algebra.support 𝒜 r ↔ graded_algebra.decompose 𝒜 r i ≠ 0 :=
by rw [graded_algebra.support, dfinsupp.mem_support_iff]
```

Reply:
Once again, `unfold graded_algebra.proj` was a big mistake!

Reply:
I guess you'll have to apply this manually, since github thinks it's outdated.

Reply:
Yes, but for some reason, I have to add a line of `simp`, otherwise, it won't compile